### PR TITLE
Add Primary Buildkite Pipeline Migration

### PR DESCRIPTION
**Pull request by markjhoy**
Migrates the primary pipeline from Jenkins to Buildkite

#### Pre-Review Checklist
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,4 +1,24 @@
...
+      - "coverage/index.html"
+  - label: ":wrench: Linter"
+    commands:
+      - "./.buildkite/scripts/run_command.sh linter"
+  - label: ":package: Docker"
+    commands:
+      - "./.buildkite/scripts/run_command.sh docker"
+  - label: ":package: Packaging"
+    commands:
+      - "./.buildkite/scripts/run_command.sh packaging"
```
Comment (acrewdson) [inline .buildkite/pipeline.yml:22]: nit: are the leading `./`s needed here?

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -1,4 +1,24 @@
-steps:
-  - label: "Placeholder Pipeline"
-    command: echo "Hello!"
+agents:
+  provider: "gcp"
+  machineType: "n1-standard-8"
+  useVault: true
```
Comment (acrewdson) [inline .buildkite/pipeline.yml:4]: just checking: do we need this? not immediately seeing where we need a Vault token in this pipeline

**Review by acrewdson — COMMENTED** [review]
looks good, in addition to a couple nits, just a few questions:

* the jenkins pipeline appears to do some artifact archiving [here](https://github.com/elastic/connectors-ruby/blob/main/.ci/pipelines/connectors.groovy#L76); we don't need that any longer?
* also looks like maybe we had [slack notifications setup](https://github.com/elastic/connectors-ruby/blob/main/.ci/pipelines/connectors.groovy#L80) (although not 100% sure they were enabled); confirming we no longer need this?

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (markjhoy) [reply .buildkite/pipeline.yml:4]: Good call - let's find out ;) incoming commit

**Comment by markjhoy** [comment]
> * the jenkins pipeline appears to do some artifact archiving [here](https://github.com/elastic/connectors-ruby/blob/main/.ci/pipelines/connectors.groovy#L76); we don't need that any longer?

The `artifact_paths` in the pipeline file will upload the artifacts that are the same from the Jenkins version (there's the coverage report in `tests` and the gemfile itself in `packaging`).

> * also looks like maybe we had [slack notifications setup](https://github.com/elastic/connectors-ruby/blob/main/.ci/pipelines/connectors.groovy#L80) (although not 100% sure they were enabled); confirming we no longer need this?

This will be enabled automagically once this gets into `main` as we have the notifications set for for all pipelines that are under the `enterprise-search` team to flow into the alerting Slack channel.

**Comment by acrewdson** [comment]
> The `artifact_paths` in the pipeline file will upload the artifacts that are the same from the Jenkins version (there's the coverage report in `tests` and the gemfile itself in `packaging`).

oh nice, totally missed that, thank you!
 
> This will be enabled automagically once this gets into `main` as we have the notifications set for for all pipelines that are under the `enterprise-search` team to flow into the alerting Slack channel.

awesome!

**Review by acrewdson — APPROVED** [review]

**Review by artem-shelkovnikov — APPROVED** [review]
Good one!

**Comment by artem-shelkovnikov** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.3|[](https://github.com/elastic/connectors-ruby/pull/590)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**